updatechecker/src/updatechecker.c

```diff
@@ -185,15 +191,30 @@ version_compare(const gchar *current_version)
 }
 
 
-static void update_check_result_cb(SoupSession *session,
-    SoupMessage *msg, gpointer user_data)
+static gchar *bytes_to_string(GBytes *bytes)
+{
+    gsize bytes_size = g_bytes_get_size(bytes);
+    gchar *str = g_malloc(bytes_size + 1);
+    memcpy(str, g_bytes_get_data(bytes, NULL), bytes_size);
```

Comment:
I don't know any version of libsoup, but wondering why wrap `g_bytes_get_data()` with another function and memcopy that has to be freed later, rather than `g_bytes_get_data()` directly, and letting `g_bytes_unref()` take care of it.

Reply:
Perhaps the response body might not be a nul terminated string, so its copied to one.

Reply:
I see... thanks.

Reply:
Yeah that's exactly the reason: the bytes have no reason to be NUL-terminated, as it's from reading the raw data from the webserver (and does not explicitly add a NUL itself from what I can gather).  So this helper is to make it a C string, not really caring if there's a NUL earlier because a valid response won't have one anyway.